### ISSUE TITLE
Error importing FIT file for a swim

### ISSUE THREAD

**Issue by zahidhanif** [jovandeginste/workout-tracker#38]
Tried to import a swimming file and got the error “ Encountered 1 problems while adding workouts: sql: converting argument $10 type: json: unsupported value: NaN”
Can you please help?

**Comment by jovandeginste** [comment]
Would you mind sharing the file?

**Comment by zahidhanif** [comment]
Sure, no problem. I have attached the file.

￼

**Comment by zahidhanif** [comment]
Sorry I replied by email. Let me try again on GH
[Tuesday Evening Pool Swim.fit.zip](https://github.com/jovandeginste/workout-tracker/files/14502260/Tuesday.Evening.Pool.Swim.fit.zip)

**Comment by zahidhanif** [comment]
This issue is back with the newer versions. When I click on the swim workout, I am getting a 500 internal error.

**Comment by jovandeginste** [comment]
Fixed again in master branch

**Comment by zahidhanif** [comment]
Thanks for the quick response!. I have pulled the latest version, deleted the workout and imported again. The import worked but when I try and rename or click link for the workout, I am still getting an internal server error.

**Comment by zahidhanif** [comment]
I pulled the latest updates and it's working now. Thanks
